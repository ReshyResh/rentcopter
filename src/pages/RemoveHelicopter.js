import { useSelector } from 'react-redux';
import { useEffect } from 'react';
import HeliCard from '../components/HeliCard';
import { topDown } from '../animations';

const RemoveHelicpoter = () => {
  window.sessionStorage.getItem("token") == null ? window.location="/login" : null
  const helicopters = useSelector((state) => state.helicopters.helicopters_arr);
  useEffect(() => {
    topDown();
  }, []);
  return (
    <div id="main" className="p-4 md:p-24 flex justify-center flex-col w-full md:w-3/4 lg:w-5/6 h-screen transition-all opacity-0 -translate-y-full origin-left ease-out duration-500">
      <h1 className="cool-title text-center !text-3xl md:!text-5xl">Delete a helicopter</h1>
      <p className="text-center mt-3">You will be able to see the helicopters in this page.</p>
      <hr className="h-1 border-slate-700 mt-4" />
      <ul className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 justify-evenly mt-20 m-auto w-full h-full" style={{ gap: '20px' }}>
        {helicopters.map((helicopter) => (
          <HeliCard key={helicopter.id} helicopter={helicopter} />
        ))}
      </ul>
    </div>
  );
};

export default RemoveHelicpoter;
