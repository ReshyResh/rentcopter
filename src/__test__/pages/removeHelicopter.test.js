import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import RemoveHelicopter from '../../pages/RemoveHelicopter';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

it('Remove helicopter page renders correctly', () => {
  const tree = renderer
    .create(<Router>
        <Provider store={store}>
            <RemoveHelicopter />
        </Provider>
    </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});