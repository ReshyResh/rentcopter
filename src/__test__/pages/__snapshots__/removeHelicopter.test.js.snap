// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove helicopter page renders correctly 1`] = `
<div
  className="p-4 md:p-24 flex justify-center flex-col w-full md:w-3/4 lg:w-5/6 transition-all opacity-0 -translate-y-full origin-left ease-out duration-500"
  id="main"
>
  <h1
    className="cool-title text-center !text-3xl md:!text-5xl"
  >
    Delete a helicopter
  </h1>
  <hr
    className="h-1 border-slate-700 mt-4"
  />
  <ul
    className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 justify-evenly mt-20 m-auto w-full h-full"
    style={
      Object {
        "gap": "20px",
      }
    }
  />
</div>
`;
